src/runtime/components/Security.tsx

```diff
@@ -0,0 +1,109 @@
+import React, {useCallback, useEffect, useMemo, useState} from 'react';
+import {Box, Button, Dialog, RadioButton, Text} from '@gravity-ui/uikit';
+import {CircleCheck} from '@gravity-ui/icons';
+
+import {V3Security} from '../../includer/models';
+import {
+    getSelectedAuth,
+    isV3SecurityApiKey,
+    isV3SecurityOAuthImplicit,
+    isV3SecurityOAuthInline,
+} from '../utils';
+
+import {Column, SecurityApiKey, SecurityOAuthImplicit, SecurityOAuthInline} from '.';
+
+export const Security: React.FC<{
+    security: V3Security[];
+    projectName: string;
+}> = ({security, projectName}) => {
+    const [isOpen, setOpen] = useState(false);
+    const [activeType, setActiveType] = useState(
+        getSelectedAuth(projectName).type ?? security[0].type,
+    );
+    const [hasAnyAuthorization, setHasAnyAuthorization] = useState(
+        Boolean(getSelectedAuth(projectName).value),
+    );
+    const activeSecurityTab = useMemo(
+        () => security.find(({type}) => activeType === type),
+        [activeType],
+    );
+    if (!activeSecurityTab) {
+        throw new Error();
+    }
+
+    useEffect(() => {
+        setActiveType(getSelectedAuth(projectName).type ?? security[0].type);
+        setHasAnyAuthorization(Boolean(getSelectedAuth(projectName).value));
+    }, [isOpen]);
```

Comment:
My only issue with this PR in its current state is that there's no centralized state controller that would eliminate the need to do ad-hoc synchronizations with `sessionStorage` like this one. I think this code would benefit greatly by having a top-level controller which would sync `sessionStorage` with local state, and would perform dumps back to it as a side-effect.

Reply:
fixed. 

I did it by analogy:
https://github.com/diplodoc-platform/openapi-extension/blob/master/src/runtime/sandbox.tsx#L16

In the prepareRequest, I get the value not from the state. In other places, a single state is used.